### PR TITLE
circular dependency followups

### PR DESCRIPTION
**Pull request by glozow**
Followups from #22677 + clean up `TestLockPointValidity`

### CODE REVIEW

**Comment by glozow** [comment]
CC @MarcoFalke, comments from https://github.com/bitcoin/bitcoin/pull/22677#pullrequestreview-820587394

File: src/txmempool.cpp [hunk 1]
```diff
@@ -650,7 +649,7 @@ void CTxMemPool::removeForReorg(CChain& chain, std::function<bool(txiter)> check
     RemoveStaged(setAllRemoves, false, MemPoolRemovalReason::REORG);
     for (indexed_transaction_set::const_iterator it = mapTx.begin(); it != mapTx.end(); it++) {
         LockPoints lp = it->GetLockPoints();
```
Comment (MarcoFalke) [inline src/txmempool.cpp:651]:
```suggestion
        const LockPoints lp{it->GetLockPoints()};
```

**Review by maflcko — APPROVED** [review]
Nice. ACK

Left one nit in case you retouch

**Comment by MarcoFalke** [comment]
Thoughts on this suggested diff?

```diff
diff --git a/src/txmempool.cpp b/src/txmempool.cpp
index 27fbb8acac..2cc0aa1e90 100644
--- a/src/txmempool.cpp
+++ b/src/txmempool.cpp
@@ -5,6 +5,7 @@
 
 #include <txmempool.h>
 
+#include <chain.h>
 #include <coins.h>
 #include <consensus/consensus.h>
 #include <consensus/tx_verify.h>
diff --git a/src/txmempool.h b/src/txmempool.h
index c6e08a3ca5..15a57de922 100644
--- a/src/txmempool.h
+++ b/src/txmempool.h
@@ -14,7 +14,6 @@
 #include <utility>
 #include <vector>
 
-#include <chain.h>
 #include <coins.h>
 #include <consensus/amount.h>
 #include <indirectmap.h>
@@ -26,12 +25,13 @@
 #include <util/epochguard.h>
 #include <util/hasher.h>
 
-#include <boost/multi_index_container.hpp>
 #include <boost/multi_index/hashed_index.hpp>
 #include <boost/multi_index/ordered_index.hpp>
 #include <boost/multi_index/sequenced_index.hpp>
+#include <boost/multi_index_container.hpp>
 
 class CBlockIndex;
+class CChain;
 class CChainState;
 extern RecursiveMutex cs_main;

**Review by maflcko — APPROVED** [review]
re-ack

**Comment by glozow** [comment]
Oops forgot the txmempool.cpp include. Added suggestions, thanks @MarcoFalke

**Review by theStack — APPROVED** [review]
Code-review ACK ddd74ff65c471c5c25815ac0643d16ea0d51f58f

File: src/txmempool.cpp [hunk 2]
```diff
@@ -649,8 +649,8 @@ void CTxMemPool::removeForReorg(CChain& chain, std::function<bool(txiter)> check
     }
     RemoveStaged(setAllRemoves, false, MemPoolRemovalReason::REORG);
     for (indexed_transaction_set::const_iterator it = mapTx.begin(); it != mapTx.end(); it++) {
-        LockPoints lp = it->GetLockPoints();
-        if (!TestLockPointValidity(chain, &lp)) {
+        const LockPoints lp{it->GetLockPoints()};
+        if (!TestLockPointValidity(chain, lp)) {
             mapTx.modify(it, update_lock_points(lp));
```
Comment (MarcoFalke) [inline src/txmempool.cpp:654]: Actually I am wondering what this code is doing at all? How is it not a noop that can be removed?

**Pull request merged**